File: TEAM B2 (Season 2)/Week 5/Musk/[정렬_가장큰수].cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>

using namespace std;
bool compare(string i, string j) {
        return j + i < i + j;
}

string solution(vector<int> numbers) {
    vector<string> strnums;
    for (auto item: numbers) { strnums.push_back(to_string(item)); }
    sort(strnums.begin(), strnums.end(), compare);
    
    string answer = "";
    for (auto item: strnums) { answer += item; }
    
    if(strnums.at(0) == "0") { return "0"; }
    return answer;
}

File: TEAM B2 (Season 2)/Week 5/Moana/[정렬_가장큰수].py
def solution(numbers):
    numbers = list(map(str, numbers))
    #원소가 0이상 1,000이하이므로 4배하고 4자리만 비교하면 됨
    #문자열이라 사전순서대로 정렬이 됨
    numbers.sort(key=lambda x: (x*4)[:4], reverse=True)

    return ''.join(numbers) if ''.join(numbers)[0] != '0' else '0'
